### PR TITLE
Add docker dev environment

### PR DESCRIPTION
**Pull request by relud**
adds devcontainer and various make commands to match antenna and socorro

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+lint: .env .docker-build  ## | Lint code.
+	docker compose run --rm --no-deps shell bin/lint.sh
+
+.PHONY: lintfix
+lintfix: .env .docker-build  ## | Reformat code.
+	docker compose run --rm --no-deps shell bin/lint.sh --fix
+
+.PHONY: test
+test: .env .docker-build  ## | Run tests.
+	docker compose run --rm shell bin/test.sh
```
Comment (willkg) [inline Makefile:55]: As a side note, I've been switching all my personal projects to using `just`:

https://just.systems/

https://github.com/willkg/everett/blob/main/justfile

It adds another requirement, but it's easier to use and better at doing the things I was using Makefiles for. I don't know whether it's easy to install/use on macOS or whether it's a viable replacement for make for our purposes. But I figured I'd mention it.

File: requirements.txt [hunk 2]
```diff
@@ -640,6 +640,10 @@ urllib3==2.2.2 \
     #   requests
     #   sentry-sdk
     #   twine
+urlwait==1.0 \
```
Comment (willkg) [inline requirements.txt:643]: If we have waitfor, we can build into it everything we're using urlwait for. urlwait is pmac's, but I don't think it's maintained anymore.

**Review by willkg — APPROVED** [review]
This seems fine. I didn't test it out figuring it's a dev environment and we can fix broken things as we discover them.

File: Makefile (hunk 1 above)
Comment (relud) [reply Makefile:55]: looks awesome, i'd love to switch. I think we should decide as a team and make the change more broadly than only this repo, so i'll bring it up in the meeting on tuesday

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -34,22 +35,18 @@ jobs:
...
         env:
           SENTRY_DSN: http://public@localhost:8090/1
           STORAGE_EMULATOR_HOST: http://localhost:8001
+          PUBSUB_EMULATOR_HOST: localhost:5010
         run: |
-          docker compose up -d fakesentry gcs-emulator
-          venv/bin/pytest tests/
+          docker compose up -d fakesentry gcs-emulator pubsub
+          # Run outside docker because we are testing the matrix python version
+          PATH="venv/bin:$PATH" bin/test.sh
```
Comment (biancadanforth) [inline .github/workflows/build.yaml:47]: Ah. At first I was wondering "what is The Matrix TM", but I see we are running the tests against multiple versions of Python: https://github.com/mozilla-services/obs-common/blob/4a78aadca50bdf890ebd85a8b0e7993d313a7f08/.github/workflows/build.yaml#L17.

File: bin/test.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# Usage: bin/run_tests.sh
+#
+# Runs tests.
+#
+# This should be called from inside a container and after the dependent
+# services have been launched. It depends on:
```
Comment (biancadanforth) [inline bin/test.sh:12]: Does this comment apply anymore if we're running this outside of a container per the "Run tests" CI job in `.github/workflows/build.yaml`?

I see `make test` still runs it from a shell in a Docker container. And the environment variables are using docker aliases for the host names... Now I'm wondering how the above CI is passing still in that case. I guess we set different env vars in that context... Ah I see that we do: https://github.com/mozilla-services/obs-common/blob/4a78aadca50bdf890ebd85a8b0e7993d313a7f08/.github/workflows/build.yaml#L40-L43

**Review by biancadanforth — COMMENTED** [review]
I know this already merged, but I had planned to look at it today, so I added a couple of comments mostly for my own understanding of what's going on here.

Works a lot smoother to test PRs like https://github.com/mozilla-services/obs-common/pull/26 in the future! Though now I know how to do that in the absence of all the docker set up.

**Pull request merged**